Neo4j.Driver/Neo4j.Driver/Internal/RetryLogic.cs

```diff
@@ -70,14 +69,25 @@ public T Retry<T>(Func<T> runTxFunc)
                 }
                 catch (Exception e) when (e.IsRetriableError())
                 {
-                    exceptions.Add(e);
+                    if (!timer.IsRunning)
+                    {
+                        timer.Start();
+                    }
+
+                    if (timer.ElapsedMilliseconds < _maxRetryTimeMs)
+                    {
+                        var delay = TimeSpan.FromMilliseconds(ComputeDelayWithJitter(delayMs));
+                        _logger?.Warn(e, $"Transaction failed and will be retried in {delay}ms.");
+                        Thread.Sleep(delay);
+                        delayMs = delayMs * _multiplier;
+                        exceptions.Add(e);
+                        continue;
+                    }
 
-                    var delay = TimeSpan.FromMilliseconds(ComputeDelayWithJitter(delayMs));
-                    _logger?.Warn(e, $"Transaction failed and will be retried in {delay}ms.");
-                    Thread.Sleep(delay);
-                    delayMs = delayMs * _multiplier;
+                    exceptions.Add(e);
+                    break;
                 }
-            } while (timer.Elapsed.TotalMilliseconds < _maxRetryTimeMs);
```

Comment:
I found it is easier to read with this do and while loop than the loop with `continue` and `break`. It looks to me the reason the new code did another run because the check of `timer.ElapsedMilliseconds` happens before sleep. So is it possible that we just keep a var before sleep and compare the var here?

Reply:
Changed the code to mostly its previous version and added a simple check to the `do-while` loop. This looks simpler, thanks!